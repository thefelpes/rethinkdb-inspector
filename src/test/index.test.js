// @flow
import inspect from '../';
import rethinkdbdash from 'rethinkdbdash';

describe('inspect', () => {
  it('should wrap .run', () => {
    const r = rethinkdbdash({
      pool: false,
    });
    inspect(r, {});
    expect(r._Term.prototype.run.name).toEqual('inspectRun');
  });

  it('should call the onQuery callback when a query is run', async () => {
    const r = rethinkdbdash({
      pool: false,
    });
    const run = jest.fn(() => Promise.resolve('a'));
    r._Term.prototype.run = run;
    const onQuery = jest.fn();
    inspect(r, {
      onQuery,
    });
    await r
      .db('test')
      .table('bla')
      .get('asdf123')
      .run();
    expect(run).toHaveBeenCalledTimes(1);
    expect(onQuery).toHaveBeenCalledTimes(1);
    expect(onQuery.mock.calls[0][0]).toMatchSnapshot();
  });
});
