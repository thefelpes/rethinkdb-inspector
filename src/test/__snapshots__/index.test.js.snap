// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inspect should call the onQuery callback when a query is run 1`] = `
"r.db(\\"test\\").table(\\"bla\\").get(\\"asdf123\\")
"
`;

exports[`inspect should call the onQueryComplete callback with timing information 1`] = `
"r.db(\\"test\\").table(\\"bla\\").get(\\"asdf123\\")
"
`;
